File: UIPeriodicTable/CPlusPlus/Resources/Components/code_uiclicklabel.cpp
// UIClickLabel application source code
#include "PeriodicApp.h"


void PeriodicApp::setup_uiclicklabel( UIWidget *layout )
{
    PODVector<UIWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp, HandleUiclicklabelEvent ));

    UIWidget *chk1 = layout->GetWidget ("somecheck");
    if (chk1)
        SubscribeToEvent(chk1, E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp, HandleUiclicklabelEvent ));

    UIWidget *chk2 = layout->GetWidget ("someradio");
    if (chk2)
        SubscribeToEvent(chk2, E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp, HandleUiclicklabelEvent ));
}

void PeriodicApp::HandleUiclicklabelEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    UIWidget* widget = static_cast<UIWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uiclicklabelcode" )
        {
            AppLog( "UIClickLabel support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Components/code_uiclicklabel.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uiclicklabellayout" )
        {
            AppLog( "UIClickLabel support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_uiclicklabel.ui.txt", widget->GetParent() );
        }

        if (widget->GetId() == "somecheck" )
        {
            AppLog( "UIClickLabel event : " + widget->GetId() + " was pressed, state = " + String ( widget->GetValue() ) );
        }
        if (widget->GetId() == "someradio" )
        {
            AppLog( "UIClickLabel event : " + widget->GetId() + " was pressed, state = " + String ( widget->GetValue() ) );
        }
    }
}

File: UIPeriodicTable/CSharp/Resources/Components/code_uiclicklabel.cs
// UIClickLabel application source code
using System;
using AtomicEngine;

public class code_uiclicklabel : CSComponent {

    public void Setup( UIWidget  layout )
    {
        var dest = new AtomicEngine.Vector<AtomicEngine.UIWidget>();
        layout.SearchWidgetClass( "TBButton", dest );
        for (var ii = 0; ii < dest.Size; ii++) {
            dest[ii].SubscribeToEvent<WidgetEvent> (dest [ii], HandleUiclicklabelEvent );
        }
        UIWidget demochk = layout.GetWidget ("somecheck");
        if ( !demochk.Equals(null))
            demochk.SubscribeToEvent<WidgetEvent> ( demochk, HandleUiclicklabelEvent );
        UIWidget demorad = layout.GetWidget ("someradio");
        if ( !demorad.Equals(null))
            demorad.SubscribeToEvent<WidgetEvent> ( demorad, HandleUiclicklabelEvent );
    }

    private static void HandleUiclicklabelEvent( WidgetEvent ev )
    {
        UIWidget widget = (UIWidget)ev.Target;
        if ( widget.Equals(null)) return;
        if ( ev.Type == UI_EVENT_TYPE.UI_EVENT_TYPE_CLICK) {
            if (widget.GetId() == "uiclicklabelcode" ) {
                AtomicMain.AppLog( "UIClickLabel support : " + widget.GetId() + " was pressed " );
                AtomicMain.ViewCode ( "Components/code_uiclicklabel.cs", widget.GetParent() );
            }
            if (widget.GetId() ==  "uiclicklabellayout" ) {
                AtomicMain.AppLog( "UIClickLabel support : " + widget.GetId() + " was pressed ");
                AtomicMain.ViewCode ( "Scenes/layout_uiclicklabel.ui.txt", widget.GetParent() );
            }

            if (widget.GetId() == "somecheck" ) {
                AtomicMain.AppLog( "UIClickLabel event : " + widget.GetId() + " was pressed, state = " +  widget.GetValue().ToString() );
            }
            if (widget.GetId() == "someradio" ) {
                AtomicMain.AppLog( "UIClickLabel event : " + widget.GetId() + " was pressed, state = " + widget.GetValue().ToString() );
            }
        }
    }
}

